contracts/TokenRequest.sol

```diff
@@ -1,161 +1,157 @@
-    string private constant ERROR_CANNOT_MINT_ZERO = "TOKENREQUEST_CANNOT_MINT_ZERO";
-    string private constant ERROR_ETH_VALUE_MISMATCH = "TOKENREQUEST_ETH_VALUE_MISMATCH";
-    string private constant ERROR_DEPOSIT_VALUE_MISMATCH = "TOKENREQUEST_DEPOSIT_VALUE_MISMATCH";
-    string private constant ERROR_DEPOSIT_VALUE_ZERO = "TOKENREQUEST_DEPOSIT_VALUE_ZERO";
-    string private constant ERROR_DEPOSIT_TOKEN_MISMATCH = "TOKENREQUEST_DEPOSIT_TOKEN_MISMATCH";
-    string private constant ERROR_MINT_VALUE_MISMATCH = "TOKENREQUEST_MINT_VALUE_MISMATCH";
-    string private constant ERROR_DEPOSIT_NOT_ACTIVE = "TOKENREQUEST_DEPOSIT_NOT_ACTIVE";
-    string private constant ERROR_TOKEN_TRANSFER_REVERTED = "TOKENREQUEST_TOKEN_TRANSFER_REVERT";
-    string private constant ERROR_TOKEN_APPROVE_FAILED = "TOKENREQUEST_TKN_APPROVE_FAILED";
-
-     struct Request {
+
+    using SafeERC20 for ERC20;
+    using UintArrayLib for uint256[];
+
+    bytes32 constant public SET_TOKEN_MANAGER_ROLE = keccak256("SET_TOKEN_MANAGER_ROLE");
+    bytes32 constant public SET_VAULT_ROLE = keccak256("SET_VAULT_ROLE");
+    bytes32 constant public FINALISE_TOKEN_REQUEST_ROLE = keccak256("FINALISE_TOKEN_REQUEST_ROLE");
+
+    string private constant ERROR_NO_AMOUNT = "TOKEN_REQUEST_NO_AMOUNT";
+    string private constant ERROR_NO_DEPOSIT = "TOKEN_REQUEST_NO_DEPOSIT";
+    string private constant ERROR_ETH_VALUE_MISMATCH = "TOKEN_REQUEST_ETH_VALUE_MISMATCH";
+    string private constant ERROR_TOKEN_TRANSFER_REVERTED = "TOKEN_REQUEST_TOKEN_TRANSFER_REVERTED";
+
+    struct TokenRequest {
+        address requesterAddress;
+        address depositToken;
         uint256 depositAmount;
-        address token;
-        uint256 mintAmount;
-        bool active;
+        uint256 requestAmount;
     }
 
-    Vault public vault;
     TokenManager public tokenManager;
-    Voting public voting;
-    MiniMeToken private token;              //temporary workaround, to show amount of tokens on radspecs's redeem function
-
-    // In case of refund how can we know which pending refund to do?
-    mapping (address => mapping (uint256 => Request)) internal requests;
-    mapping (address => uint256) public requestsLengths;
-
-    event Request(address indexed receiver, address depositToken, uint256 depositAmount, uint256 mintAmount, uint256 voteId, uint256 requestId);
-    event ApprovedRequest(
-        address indexed receiver,
-        address depositToken,
-        uint256 depositAmount,
-        uint256 mintAmount,
-        uint256 voteId,
-        uint256 requestId
-        );
-
-     /**
-    * @notice Initialize TokenRequest app contract
-    * @param _vault Address of the vault
-    * @param _tokenManager TokenManager address
-    * @param _voting Address of the voting
-    */
-    function initialize(Vault _vault, TokenManager _tokenManager, Voting _voting) external onlyInit {
+    address public vault;
```

Comment:
We have no use for the Vault functionality in this contract. We could opt to use the `deposit()` function on this address/Vault, but then we would have to do an `approve()` before attempting to deposit any ERC20 tokens, which I think is unnecessary. And it would restrict the benefactor address to being of type Vault, and maybe someone deploying this would rather use something other than a Vault.